pkg/bypass4netns/bypass4netns.go

```diff
@@ -0,0 +1,585 @@
+	syscallName, err := ctx.req.Data.Syscall.GetName()
+	if err != nil {
+		logrus.Errorf("Error decoding syscall %v(): %s", ctx.req.Data.Syscall, err)
+		// TODO: error handle
+		return
+	}
+	logrus.Debugf("Received syscall %q, pid %v, arch %q, args %+v", syscallName, ctx.req.Pid, ctx.req.Data.Arch, ctx.req.Data.Args)
+
+	ctx.resp.Flags |= C.SECCOMP_USER_NOTIF_FLAG_CONTINUE
+
+	switch syscallName {
+	case "connect":
+		handleSysConnect(ctx, opts)
+	case "bind":
+		handleSysBind(ctx, opts)
+	case "setsockopt":
+		handleSysSetsockopt(ctx, opts)
+	case "close":
+		// handling close(2) may cause performance degradation
+		handleSysClose(ctx, opts)
+	default:
+		logrus.Errorf("Unknown syscall %q", syscallName)
+		// TODO: error handle
+		return
+	}
+
+}
+
+// notifHandler handles seccomp notifications and response to them.
+func notifHandler(fd libseccomp.ScmpFd, metadata string) {
+	defer unix.Close(int(fd))
+	opts := socketOptions{
+		options: map[string][]socketOption{},
+	}
+
+	for {
+		req, err := libseccomp.NotifReceive(fd)
+		if err != nil {
+			logrus.Errorf("Error in NotifReceive(): %s", err)
+			continue
+		}
+
+		ctx := context{
+			notifFd: fd,
+			req:     req,
+			resp: &libseccomp.ScmpNotifResp{
+				ID:    req.ID,
+				Error: 0,
+				Val:   0,
+				Flags: libseccomp.NotifRespFlagContinue,
+			},
+		}
+
+		// TOCTOU check
+		if err := libseccomp.NotifIDValid(fd, req.ID); err != nil {
+			logrus.Errorf("TOCTOU check failed: req.ID is no longer valid: %s", err)
+			continue
+		}
+
+		handleReq(&ctx, &opts)
+
+		if err = libseccomp.NotifRespond(fd, ctx.resp); err != nil {
+			logrus.Errorf("Error in notification response: %s", err)
+			continue
+		}
+	}
+}
+
+type Handler struct {
+	socketPath string
+}
+
+// NewHandler creates new seccomp notif handler
+func NewHandler(socketPath string) *Handler {
+	handler := Handler{
+		socketPath: socketPath,
+	}
+
+	return &handler
+}
+
+// StartHandle starts seccomp notif handler
+func (h *Handler) StartHandle() {
+	logrus.Info("Waiting for seccomp file descriptors")
+	l, err := net.Listen("unix", h.socketPath)
+	if err != nil {
+		logrus.Fatalf("Cannot listen: %s", err)
+	}
+	defer l.Close()
+
+	for {
+		conn, err := l.Accept()
+		if err != nil {
+			logrus.Errorf("Cannot accept connection: %s", err)
+			continue
+		}
+		socket, err := conn.(*net.UnixConn).File()
+		conn.Close()
+		if err != nil {
+			logrus.Errorf("Cannot get socket: %v", err)
+			continue
+		}
+		newFd, metadata, err := handleNewMessage(int(socket.Fd()))
+		socket.Close()
+		if err != nil {
+			logrus.Errorf("Error receiving seccomp file descriptor: %v", err)
+			continue
+		}
+
+		// Make sure we don't allow strings like "/../p", as that means
+		// a file in a different location than expected. We just want
+		// safe things to use as a suffix for a file name.
+		metadata = filepath.Base(metadata)
+		if strings.Contains(metadata, "/") {
+			// Fallback to a safe string.
+			metadata = "agent-generated-suffix"
+		}
```

Comment:
The metadata logic is specific to the mkdir example from runc, so we can remove this